src/VisualStudio/Core/Def/Packaging/PackageSearchService.cs

```diff
@@ -94,12 +91,49 @@ private static IPackageSearchRemoteControlService CreateRemoteControlService(VSS
 
             _cacheDirectoryInfo = new DirectoryInfo(Path.Combine(
                 localSettingsDirectory, "PackageCache", string.Format(Invariant($"Format{_dataFormatVersion}"))));
-            // _databaseFileInfo = new FileInfo(Path.Combine(_cacheDirectoryInfo.FullName, "NuGetCache.txt"));
 
             _cancellationTokenSource = cancellationTokenSource;
             _cancellationToken = _cancellationTokenSource.Token;
         }
 
+        internal void Start()
+        {
+            var options = _workspace.Options;
+            if (!options.GetOption(ServiceComponentOnOffOptions.PackageSearch))
+            {
+                return;
+            }
+
+            // Start the whole process once we're connected
+            _installerService.PackageSourcesChanged += OnPackageSourcesChanged;
+            OnPackageSourcesChanged(this, EventArgs.Empty);
+            _started = true;
```

Comment:
Should this happen before the call to `OnPackageSourcesChanged` in case anything in that method looks at it?